File: georgia_tech/sgl-eval-peterhj-genserver/backend/eval_next_backend/api_types/benchmarks.py
from typing import List, Dict
from pydantic import BaseModel

class BenchmarkMeta(BaseModel):
    id: str
    name: str
    description: str

class BenchmarkOptionsResponse(BaseModel):
    benchmarks: List[BenchmarkMeta]

class LlmMeta(BaseModel):
    id: str
    name: str

class BenchmarkMetadataResponse(BaseModel):
    benchmarks: List[BenchmarkMeta]
    llms: List[LlmMeta]

class BenchmarkModelMeta(BaseModel):
    score: float

class BenchmarkTableRow(BaseModel):
    id: int
    question: str
    difficulty: float
    model_metas: Dict[str, BenchmarkModelMeta]

class BenchmarkResponse(BaseModel):
    table_rows: List[BenchmarkTableRow] 

File: georgia_tech/sgl-eval-peterhj-genserver/website/src/api_types/benchmarks.ts
export type BenchmarkId = string;

export type BenchmarkMeta = {
  id: BenchmarkId;
  name: string;
  description: string;
};

export type LlmMeta = {
  id: string;
  name: string;
};

export type BenchmarkMetadataResponse = {
  benchmarks: BenchmarkMeta[];
  llms: LlmMeta[];
};

export type BenchmarkModelMeta = {
  score: number;
};

export type BenchmarkTableRow = {
  id: number;
  question: string;
  difficulty: number;
  model_metas: Record<string, BenchmarkModelMeta>;
};

export type BenchmarkResponse = {
  table_rows: BenchmarkTableRow[];
}; 